test/spiretest/socketapi.go

```diff
@@ -60,6 +67,11 @@ func StartGRPCSocketServer(t *testing.T, socketPath string, registerFn func(s *g
 	}()
 	t.Cleanup(func() {
 		server.Stop()
-		require.NoError(t, <-errCh)
+		err := <-errCh
+		switch {
```

Comment:
nit:
```
switch err {
case nil, grpc.ErrServerStopped:
default:
    t.Fatal(err)
}
```
i'd even prefer putting a `return` in the `case` statement to be more explicit about what's happening, but up to you